plugin/trino-mongodb/src/main/java/io/trino/plugin/mongodb/expression/ContainsRewriteRule.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.mongodb.expression;
+
+import io.trino.matching.Captures;
+import io.trino.matching.Pattern;
+import io.trino.matching.Property;
+import io.trino.plugin.base.expression.ConnectorExpressionRule;
+import io.trino.plugin.mongodb.MongoColumnHandle;
+import io.trino.spi.expression.Call;
+import io.trino.spi.expression.ConnectorExpression;
+import io.trino.spi.expression.Constant;
+import io.trino.spi.expression.FunctionName;
+import io.trino.spi.expression.Variable;
+import io.trino.spi.type.ArrayType;
+import io.trino.spi.type.Type;
+import org.bson.Document;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.ZoneOffset;
+import java.util.Date;
+import java.util.List;
+import java.util.Optional;
+
+import static io.trino.plugin.mongodb.TypeUtils.isPushdownSupportedArrayElementType;
+import static io.trino.plugin.mongodb.TypeUtils.translateValue;
+
+public class ContainsRewriteRule
+        implements ConnectorExpressionRule<Call, Document>
+{
+    private final Pattern<Call> expressionPattern;
+
+    public ContainsRewriteRule()
+    {
+        Pattern<Call> p = Pattern.typeOf(Call.class);
```

Comment:
Maybe `pattern` instead of `p` - It can also be a static final variable right ?

Reply:
done